File: public/usage-examples/animations/assign_animation_with_script_named-1-example.cpp
#include "splashkit.h"

int main()
{
    // Look up the script by name and assign an animation to WalkFront
    animation_anim anim = create_animation();
    animation_script script = animation_script_named("WalkingScript");

    assign_animation(anim, script, "WalkFront");
    write_line("Assigned WalkFront -> script name: %s", animation_script_name(script).c_str());

    free_animation(anim);
    free_animation_script(script);
    return 0;
}

File: public/usage-examples/animations/assign_animation_with_script_named-1-example.py
from splashkit import *

# Use the script name to retrieve the script, then assign the WalkFront animation
anim = create_animation()
script = animation_script_named("WalkingScript")

assign_animation_with_script_named(anim, "WalkingScript", "WalkFront")
write_line(f"Assigned WalkFront -> script name: {animation_script_name(script)}")

free_animation(anim)
free_animation_script(script)
